java-rmi/src/test/java/com/baeldung/rmi/JavaRMIIntegrationTest.java

```diff
@@ -0,0 +1,45 @@
+package com.baeldung.rmi;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.fail;
+
+import java.rmi.NotBoundException;
+import java.rmi.RemoteException;
+import java.rmi.registry.LocateRegistry;
+import java.rmi.registry.Registry;
+
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+public class JavaRMIIntegrationTest {
+	
+	@BeforeClass
+	public static void whenRunServer_thenServerStarts() {
+		
+		try {
+			MessengerServiceImpl server = new MessengerServiceImpl();
+			server.createStubAndBind();
+		} catch (RemoteException e) {
+			e.printStackTrace();
+			fail("Exception Occured");
+		}
+	}
+	
+	@Test
+	public void whenClientSendsMessageToServer_thenServerSendsResponseMessage() {
+		
+		try {
+			Registry registry = LocateRegistry.getRegistry();	
+			MessengerService server = (MessengerService) registry.lookup("MessengerService");			
+			String responseMessage = server.sendMessage("Client Message");		
+			
+			String expectedMessage = "Server Message";			
+			assertEquals(responseMessage, expectedMessage);
+		} catch (RemoteException e) {
+			fail("Exception Occured");
```

Comment:
Sorry to add one more thing: I just noticed this that Occurred is mispelled. It should be "Occurred" instead of "Occured". This is in multiple places. I only mention it here since you are already making a couple of other changes for the unit test.